File: wrappers/python/src/enums.cpp
#include <boost/python/enum.hpp>
#include <boost/python/module.hpp>
#include <enums.hpp>

using namespace boost::python;

BOOST_PYTHON_MODULE(_testmatch) {
    enum_<Arm>("Arm").value("left", left).value("right", right).export_values();

    enum_<BowlType>("BowlType")
        .value("med", med)
        .value("med_fast", med_fast)
        .value("fast_med", fast_med)
        .value("fast", fast_med)
        .value("offbreak", offbreak)
        .value("legbreak", legbreak)
}

File: src/testmatch/enums.py
from enum import Enum


class Arm(str, Enum):
    left = "left"
    right = "right"


class BowlType(str, Enum):
    med = "med"
    med_fast = "med_fast"
    fast_med = "fast_med"
    fast = "fast"
    offbreak = "offbreak"
    legbreak = "legbreak"


class DismType(str, Enum):
    bowled = "bowled"
    lbw = "lbw"
    caught = "caught"
    c_and_b = "c_and_b"
    run_out = "run_out"
    stumped = "stumped"
